### PR TITLE
Fix Advanced indexing of Cupy's __getitem__ when an index argument is a list

### PR DESCRIPTION
**Pull request by yuyu2172**
This PR fixes a bug found in #2418.

### CODE REVIEW

File: cupy/core/core.pyx [hunk 1]
```diff
@@ -1098,10 +1098,18 @@ cdef class ndarray:
...
+        if isinstance(slices, tuple):
             slices = list(slices)
+        elif isinstance(slices, list):
+            slices = list(slices)  # copy list
+            single_integer_array_indexing = True
+            for s in slices:
+                if not isinstance(s, int):
+                    single_integer_array_indexing = False
+            if single_integer_array_indexing:
+                slices = [slices]
```
Comment (niboshi) [inline cupy/core/core.pyx]: How about replacing this part with `if all(isinstance(s, int) for s in slices)`? It's more idiomatic and readable. Your code is slightly faster, but the difference is negligible.
The same goes for PR #2420.

**Comment by niboshi** [comment]
Thank you very much for the PR!
Could you consider my review comment?

**Comment by yuyu2172** [comment]
Thanks for commenting.
I fixed both this PR and #2420.

**Comment by niboshi** [comment]
LGTM. Thank you!

**Pull request merged**